### PR TITLE
New feature/data organization

### PR DESCRIPTION
**Pull request by tmb7852**

### CODE REVIEW

**Review by EvanHughes-dev — APPROVED** [review]

File: DDL.sql [hunk 1]
```diff
@@ -121,27 +128,7 @@ CREATE TABLE BmsBatteryVoltage(
...
-    FOREIGN KEY (contextId) REFERENCES Context(id)
-);
-
-CREATE TABLE BmsMaxCellVoltage(
-    id          SERIAL PRIMARY KEY,
-    packId      INTEGER NOT NULL,
-    cellId      INTEGER NOT NULL,
-    val         INTEGER NOT NULL,
-    receiveTime TIMESTAMP NOT NULL,
+    receiveTime DECIMAL(11,6) NOT NULL,
```
Comment (tmb5932) [inline DDL.sql:131]: Whats reasoning behind using a decimal value instead of a timestamp for a time?

File: DDL.sql [hunk 2]
```diff
@@ -281,16 +248,18 @@ CREATE TABLE TmuThermistorError(
...
     contextId   INTEGER NOT NULL,
     FOREIGN KEY (contextId) REFERENCES Context(id)
 );
 
 CREATE TABLE TmsSensorTemp(
     id          SERIAL PRIMARY KEY,
+--     sensorId  VARCHAR(30),
+--     CONSTRAINT DSI CHECK (sensorId IN ('Between Motor Inverter', 'Radiator Fan 1', 'Radiator Fan 2','TMS Internal')),
```
Comment (tmb5932) [inline DDL.sql:259]: should the commented out attributes & contraints be deleted?

File: TelemetrySite/server/data_organization.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+#
+# @param tpdo_config the individual config for the tpdo being handled
+# @param data_bytes the bytes of data from the can message
+# @param recieveTime the timestamp of the data as a decimal value
+# @param context_id the context in which the data is being recorded
+def upload_organized(tpdo_config, data_bytes, receive_time, context_id):
+    conn = utils.connect()
+    cur = conn.cursor()
+    binary_data_string = ""
+    # Convert the bytes into a binary string (This makes it 8x biigger than before!)
```
Comment (tmb5932) [inline TelemetrySite/server/data_organization.py:97]: wow its so much biigger!!!

File: DDL.sql (hunk 1 above)
Comment (tmb5932) [reply DDL.sql:131]: Trever told me and it makes sense. 🐸

**Review by tmb5932 — APPROVED** [review]
Just some small things left (mainly comment stuff) after I talked w Trever about my comments. I trust the DATABASE-MAN to fix them 🐸

File: TelemetrySite/server/data_organization.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            if(index != "table" and index != "size"):
+                insert_string += "%s, "
+                column_string += "{}, "
+                insert_columns.append(index)
+                data_insert_array += str(data_package[index])
+        insert_string += "%s, %s, %s"
+        column_string += "{}, {}, {}"
+        insert_columns += ["val", "receiveTime", "contextId"]
+        data_insert_array += [insert_data_int, str(receiveTime), contextId]
+        finalized_sql_insert_string_tuple = data_insert_array
```
Comment (Ethan-Caracoglia) [inline TelemetrySite/server/data_organization.py]: Holy variable name

**Comment by Ethan-Caracoglia** [comment]
No issues as far as I can see and every changes seems to make sense

**Review by Ethan-Caracoglia — APPROVED** [review]
Everything looks good or at the very least makes sense :)

File: TelemetrySite/server/data_organization.py (hunk 4 above)
Comment (tmb7852) [reply TelemetrySite/server/data_organization.py]: I am willing to accept alternate ideas

**Pull request merged**